File: Sen/Remove_Linked_List_Elements.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def removeElements(self, head, val):
        """
        :type head: ListNode
        :type val: int
        :rtype: ListNode
        """
        dummy = ListNode(0)
        pre = dummy
        pre.next = head
        
        cur = head
        while cur:
            if cur.val == val:
                # cur = cur.next 
                pre.next = pre.next.next
            else:
                # cur = cur.next
                pre = pre.next 
            cur = cur.next 
        head = dummy.next 
        return head

File: xiaoming/1011/RemoveLinkedListElements.java

public class RemoveLinkedListElements {
	
	public ListNode removeElements(ListNode head, int val) {
        
        while (head != null && head.val == val) {
            head = head.next;
        }
        ListNode pre = new ListNode(0);
        pre.next = head;
        while (head.next != null) {
            if(head.next.val == val)
               head.next = head.next.next;
            else head = head.next;
        }
        
        return pre.next;
    }
	

}


class ListNode {
	     int val;
	      ListNode next;
	     ListNode(int x) { val = x; }
	  }